argocd/resource_argocd_application_test.go

```diff
@@ -1659,6 +1694,60 @@ resource "argocd_application" "crds" {
 	`, name)
 }
 
+func testAccArgoCDApplicationCustomNamespace(name string) string {
+	return fmt.Sprintf(`
+resource "argocd_project" "simple" {
+  metadata {
+    name      = "%[1]s"
```

Comment:
:thinking: Not sure I follow you here. There are 3 parameters in the format string. The 1st and 3rd are the project name (and so should be the same) whereas the 2nd is the application name. So, I could change this to something like 

``` go
fmt.Sprintf(`
resource "argocd_project" "simple" {
  metadata {
    name      = "%[1]s"
...
resource "argocd_application" "simple" {
  metadata {
    name      = "%[2]s
    namespace = "mynamespace-1"
  }
  spec {
    project = "%[1]s"
...
`, projectName, applicationName)
```

Whereas, if I were to change to using `%s` then I would need to repeat the `name` parameter it would look something like the following (unless there's some syntax that I'm missing here?)

``` go
fmt.Sprintf(`
resource "argocd_project" "simple" {
  metadata {
    name      = "%s"
...
resource "argocd_application" "simple" {
  metadata {
    name      = "%s
    namespace = "mynamespace-1"
  }
  spec {
    project = "%s"
...
`, name, name, name)
```

Also, the option using `%s` is now strictly incorrect as parameter 1 and 3 could be different. And so, since we don't really care if the project name and the application name are different, I took the easy (lazy?) way out and opted for the more terse form of making them all the same rather than option 1. Happy to adjust to either if you would still prefer that over the current implementation?

Reply:
I fully agree, but in this case of the `func testAccArgoCDApplicationCustomNamespace(name string) string
` function (line 1697), only _one_ parameter is provided: `name`. Therefore format indexing is indeed not needed ;)

Reply:
> I didn't fully read your comment, my sincerest apologies. It was indeed more satisfactory with the %[1]s notation, feel free to roll this change back. Again my apologies! bow

You did have me very confused :smile: No worries. Have dropped that commit now.